### PR TITLE
Rev 82fb373 medallia integration

### PR DESCRIPTION
**Pull request by axqx**

### CODE REVIEW

File: controller.go [hunk 1]
```diff
@@ -394,6 +394,12 @@ func (c *controller) ReloadConfiguration(cfgOptions ...config.Option) error {
 	c.cfg = cfg
 	c.Unlock()
 
+	if c.discovery == nil && c.cfg.Cluster.Watcher != nil {
```
Comment (mauri) [inline controller.go]: not part of our changes (see https://github.com/medallia/libnetwork/commit/9876affaed02879b20a954b9fc9c75b32e513308)

File: drivers/overlay/ov_network.go [hunk 2]
```diff
@@ -625,6 +625,7 @@ func (n *network) watchMiss(nlSock *nl.NetlinkSocket) {
 			if neigh.IP.To4() == nil {
 				continue
 			}
+			logrus.Debugf("miss notification for dest IP, %v", neigh.IP.String())
```
Comment (mauri) [inline drivers/overlay/ov_network.go]: harmless, but let's remove it.

File: endpoint.go [hunk 3]
```diff
@@ -536,6 +536,9 @@ func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) error {
 			}
 		}
 
+		if sb.resolver != nil {
```
Comment (mauri) [inline endpoint.go]: remove

File: endpoint.go [hunk 4]
```diff
@@ -616,6 +619,10 @@ func (ep *endpoint) Leave(sbox Sandbox, options ...EndpointOption) error {
 	sb.joinLeaveStart()
 	defer sb.joinLeaveEnd()
 
+	if sb.resolver != nil {
```
Comment (mauri) [inline endpoint.go]: remove

File: resolver.go [hunk 5]
```diff
@@ -30,6 +30,9 @@ type Resolver interface {
 	// SetExtServers configures the external nameservers the resolver
 	// should use to forward queries
 	SetExtServers([]string)
+	// FlushExtServers clears the cached UDP connections to external
+	// nameservers
+	FlushExtServers()
```
Comment (mauri) [inline resolver.go]: remove, I think all the additions in the file needs to go away.

File: controller.go (hunk 1 above)
Comment (axqx) [reply controller.go]: Commit a79cf19f8ed7d421efc8c1d48bfd80fbd23e207c
Done.

File: drivers/overlay/ov_network.go (hunk 2 above)
Comment (axqx) [reply drivers/overlay/ov_network.go]: Commit a79cf19f8ed7d421efc8c1d48bfd80fbd23e207c
Done.

File: endpoint.go (hunk 3 above)
Comment (axqx) [reply endpoint.go]: Commit a79cf19f8ed7d421efc8c1d48bfd80fbd23e207c
Done.

File: endpoint.go (hunk 4 above)
Comment (axqx) [reply endpoint.go]: Commit a79cf19f8ed7d421efc8c1d48bfd80fbd23e207c
Done.

File: resolver.go (hunk 5 above)
Comment (axqx) [reply resolver.go]: Commit a79cf19f8ed7d421efc8c1d48bfd80fbd23e207c
Done.

**Comment by mauri** [comment]
LGTM

**Pull request merged**